addon-test-support/setup-link.ts

```diff
@@ -12,10 +13,18 @@ export default function setupLink(hooks: NestedHooks) {
       router._router._routerMicrolib == null
     );
 
-    const linkManager = this.owner.lookup(
-      'service:link-manager'
-    ) as LinkManagerService;
+    assert(
+      'ember-link.setupLink: You have already called `setupLink` once',
+      !(
+        this.owner.lookup('service:link-manager') instanceof
+        TestInstrumentedLinkManagerService
+      )
```

Comment:
This way the actual service is not initialized unnecessarily.

```suggestion
      !this.owner.hasRegistration('service:link-manager') ||
      !(
        this.owner.lookup('service:link-manager') instanceof
        TestInstrumentedLinkManagerService
      )
```